project/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -85,7 +85,7 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
       Entity election = electionQueryResult.asSingleEntity();
       List<DirectoryCandidate> candidates =
           extractCandidateInformation((List<String>) election.getProperty("candidateIds"),
-                                      (List<String>) election.getProperty("candidateIncumbency"));
+                                      (List<Boolean>) election.getProperty("candidateIncumbency"));
       List<Position> positions =
           extractPositionInformation((List<String>) election.getProperty("candidatePositions"),
                                      candidates);
```

Comment:
Instead of doing this in 2 steps, why not just have one function that (internally) constructs a map<position name, list of candidates>?

Then `extractCandidateInformation` could instead just deal with a single candidate and you call that for each loop.

Reply:
Got it! Just to confirm, by "internally" are you referring to a helper function in DataServlet? Would removing the Position class be good? The mapping would represent all information represented in Position.

Reply:
You can probably put it all in `extractPositionInformation`. Or sure, removing position could work as well~

Reply:
Got it, thanks! I'll refactor the code for `extractPositionInformation` and `extractCandidateInformation`
I thought you meant a literal map<position name, list of candidates>. If not, I can just keep the Position class, since it is functionally equivalent but gives the mapping a more meaningful name